pkg/util/fs_util.go

```diff
@@ -247,6 +287,15 @@ func checkWhiteouts(path string, whiteouts map[string]struct{}) bool {
 	return false
 }
 
+func checkSymlinks(path string, symlinks map[string]struct{}) bool {
```

Comment:
You're right, thanks for pointing that out! I'll remove it.

Reply:
I think we might actually need the symlink check, since we only want to specially hardlink at the end of extraction if the linked file exists but can't be hardlinked (which I think only happens if it's a symlink)

If a file was normally extracted in a previous layer than we can extract the file and create the hardlink normally